pkg/rpc/client.go

```diff
@@ -0,0 +1,60 @@
+package rpc
+
+import (
+	"context"
+
+	"github.com/dusk-network/dusk-blockchain/pkg/p2p/wire/topics"
+	"github.com/dusk-network/dusk-blockchain/pkg/util/nativeutils/rpcbus"
+	"github.com/dusk-network/dusk-protobuf/autogen/go/phoenix"
+	"google.golang.org/grpc"
+)
+
+// Client is a wrapper for a gRPC client. It establishes connection with
+// the server on startup, and then handles requests from other components
+// over the RPCBus.
+type Client struct {
+	phoenix.RuskClient
+	validateSTChan chan rpcbus.Request
+	executeSTChan  chan rpcbus.Request
+}
+
+// InitRuskClient opens the connection with the Rusk gRPC server, and
+// launches a goroutine which listens for RPCBus calls that concern
+// the Rusk server.
+//
+// As the Rusk server is a fundamental part of the node functionality,
+// this function will panic if the connection can not be established
+// successfully.
+func InitRuskClient(address string, rpcBus *rpcbus.RPCBus) {
+	conn, err := grpc.Dial(address, grpc.WithInsecure(), grpc.WithBlock())
```

Comment:
This will block infinitely if Rust server is down/not responding. We can use dial timeout here and then panic?

Reply:
Good point